### ISSUE TITLE
Docs: problems with the two off-canvas components for docs navigation and general navigation on small screen

### ISSUE THREAD

**Issue by patrickhlauke** [twbs/bootstrap#36596]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Split from https://github.com/twbs/bootstrap/issues/36586 as I might have misunderstood what that issue was about, and proceeded to look at https://getbootstrap.com instead...and came across issues there:

it seems that for the navigation, something's not quite right in how the off-canvas was implemented: while the off-canvas for the "Toggle docs navigation" correctly traps focus, the off-canvas for the general navigation doesn't...users can still tab out of this and into the underlying page.

![screenshot of open navigation off-canvas, but focus is visibly in the underlying page behind the semi-transparent overlay](https://user-images.githubusercontent.com/895831/174197948-de67d48d-14c6-42cc-baa3-6effe3f5dd9f.png)

@mdo worth checking how this was added/implemented in this case. Also yes, while the "Toggle navigation" button does have an `aria-expanded`, it's currently always set to `false`, regardless. I suspect this was a left-over from the previous design that used dropdowns, as the off-canvas doesn't set `aria-expanded` since the idea is that it's a modal, and the underlying page should not actually be accessible (because of the focus trapping and the `aria-modal="true"`).

Also noticed that focus is not moved to either of the off-canvases (the docs nav, and the general nav), but remains on the toggle buttons. Not looked at the markup in depth, but suspect a `tabindex="-1"` is missing on them (compare to the working off-canvas examples https://getbootstrap.com/docs/5.2/components/offcanvas/).

### Reduced test cases

N/A

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.2

**Comment by aman-at8** [comment]
hey @patrickhlauke I would like to help on this issue. I am relatively new to open source contribution, can you please guide me.

**Comment by julien-deramond** [comment]
> hey @patrickhlauke I would like to help on this issue. I am relatively new to open source contribution, can you please guide me.

Hey @aman-at8. If you have an idea on how to fix this issue, feel free to open a PR against the _main_ branch. (see [Contributing > Pull Requests](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md#pull-requests))

**Comment by GeoSot** [comment]
@patrickhlauke it caused beacuse of
```html
<div class="offcanvas-lg offcanvas-end flex-grow-1 show" id="bdNavbar" aria-labelledby="bdNavbarOffcanvasLabel" data-bs-scroll="true" aria-modal="true" role="dialog">
```
According to code:
https://github.com/twbs/bootstrap/blob/2bf383c59640a0c693a4ab6b5dc58104c5ec2294/js/src/offcanvas.js#L115-L122
Focustrap is activated only when option `scroll= false`

**Comment by floss-dev** [comment]
Hello

With that modification focustrap is activated when scroll is disabled **or** when it has backdrop (the last condition is the new one)

Previous code was (v5.1.3):
https://github.com/twbs/bootstrap/blob/1a6fdfae6be09b09eaced8f0e442ca6f7680a61e/js/src/offcanvas.js#L120-L123

New code (from v5.2.0):
https://github.com/twbs/bootstrap/blob/edf9c40956d19e6ab3f9151bfe0dfac6be06fa21/js/src/offcanvas.js#L116-L119

This change breaks this use case (working in previous versions):

1. An offcanvas is used to show a form on it 
    - 'scroll=true' to assure 'focustrap.activate()' is not called
    -  backdrop=true, to assure other elements outside the offcanvas are selected by user
2. Some action in that form opens a pop-up (to gather more info/data)
    - Form elements in that pop-up were accesible to user (get focus, write...)

With current modification, input elements in that pop-up NEVER can get focus, becasue offcanvas trapped interaction when called: `this._focustrap.activate()`

If previous behavior can't be recovered... Could be possible to publish a property ("focustrap") to allow developer to choose if oncanvass performs that logic? ('this._focustrap.activate()/deactivate()' )
Other options are welcome

Regards

**Comment by patrickhlauke** [comment]
> Some action in that form opens a pop-up (to gather more info/data)

just on that point, i'll note that offcanvas is an offshoot of a modal, and bootstrap doesn't support modals called from within modals by default

**Comment by floss-dev** [comment]
Yes @patrickhlauke, that seems the be the new behaviour from v5.2.0
Meanwhile, previous developments/usages are impacted by the change (almost, when upgrading Bootstrap)

I think some level of backwards compatibility could be allowed:
  + new property as proposed in previous comment
  + publish functions to manipulate focustrap (activate/deactivate)
  + others…

Regards
